### ISSUE TITLE
[a11y]: MultiSelect [default] - delete selected options jaws bug

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#15257]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.39.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.39.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Description
JAWS behavior with deletion of selected options is not intuitive to the users

1. As per the instruction, when the user presses "back space" or "delete", SR does not announce any feedback. Hence the user would not know the confirmation.
2. Pressing "enter" or "spacebar" toggles the selection and the same is announced by the SR instentaniously. However the way of announcement may cause some confusion for the user.

### Recommended Fix
1. When the selection status changes with "Enter" and "Spacebar" action, The currently selected option can be given the ARIA selected attribute. or the word "selected" can be added along with the label and updated according to the toggle state.
2. Need confirmation on the expected behavior on "Back Space" and "delete" actions

When the state of an element i.e. disabled, selected, expanded, currently active etc. is not conveyed or conveyed inappropriately for the screen reader user, it can be misleading for the user. . this impacts the user interaction.

### WCAG 2.1 Violation
4.1.2 Name, Role, Value
